capstone/src/main/java/com/google/univiz/scorecard/CollegeIdReaderProviderImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.google.univiz.scorecard;
+
+import com.google.univiz.api.CollegeId;
+import com.google.univiz.config.UnivizConfig;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import javax.inject.Inject;
+
+class CollegeIdReaderProviderImpl implements CollegeIdReaderProvider {
```

Comment:
This seam that you created for testing is awesome - but it still has a lot of untested magic in it:
* Making sure that the constructed URL string is correct
  - has the right API key, and IDs
* that an inputstreamreader is wrapping the result (although I'm less worried about this)

a couple of options:
- Change the return type of the seam to InputStream
- Change the input to the provider to be a String - then you can verify that the String is computed correctly in the client of this provider

Also - can you think of a way to unit test this implementation (without connecting to a remote REST resource)?

Feel free to chat with me offline about this comment.

Reply:
@mmurray22 - this seam will be useful for #83, as well. Perhaps you can share this as common infrastructure?

Reply:
+1 You can probably just move getReaderFromCollegeIds back to CollegeDataApiImpl so it can be tested.